**Note on CI failure:** The test failure is pre-existing and unrelated to this PR. `test_mq_websocket_api.py` fails during collection because `ovos_utils.signal` was removed upstream in a newer `ovos-utils` release:

```
neon_utils/file_utils.py:37: in <module>
    from ovos_utils.signal import ensure_directory_exists
E   ModuleNotFoundError: No module named 'ovos_utils.signal'
```

The import chain is `test_mq_websocket_api.py` → `MQWebsocketAPI` → `neon_iris.client` → `neon_utils.file_utils` → `ovos_utils.signal`. Last successful `dev` CI run was February 26 — the dependency break happened sometime after that.

All 45 tests in this PR pass locally in Docker (`ghcr.io/neongeckocom/neon-hana:dev`).

--- a/neon_hana/app/routers/hub.py
+++ b/neon_hana/app/routers/hub.py
@@ -29,6 +29,7 @@ from typing import Optional
 
 import yaml
 from fastapi import APIRouter, Depends
+from neon_data_models.models.api.llm import LLMPersona
 from ovos_config.config import update_mycroft_config
 from ovos_utils.log import LOG
 
@@ -37,7 +38,6 @@ from neon_hana.hub_id import generate_hub_id
 from neon_hana.schema.hub_requests import (
     HubConfigResponse,
     HubIdentityResponse,
-    LLMConfig,
     STTConfig,
     TTSConfig,
     UpdateHubIdentityRequest,
@@ -136,6 +136,11 @@ async def get_hub_config() -> HubConfigResponse:
     This endpoint is public (no authentication required) so that
     Nodes can display Hub configuration during discovery.
     """
+    # TODO: This relies on HANA sharing a local FS with the core services
+    # (true for a standard Hub deployment, not for split/cloud deployments).
+    # When opm.tts.query / opm.stt.query are reliably handled by ovos-audio
+    # / neon-speech, fall back to messagebus queries via MQ for non-Hub
+    # deployments. See the related research note in the project tracker.
     neon_config = _read_neon_yaml()
 
     if neon_config is None:
@@ -150,5 +155,8 @@ async def get_hub_config() -> HubConfigResponse:
     return HubConfigResponse(
         tts=tts,
         stt=stt,
-        llm=LLMConfig(name="Neon Classic"),
+        llm=LLMPersona(
+            name="Neon Classic",
+            description="OVOS parser-based assistant (no LLM)",
+        ),
     )

--- a/neon_hana/app/routers/hub.py
+++ b/neon_hana/app/routers/hub.py
@@ -28,8 +28,10 @@ import os
 from typing import Optional
 
 import yaml
-from fastapi import APIRouter, Depends
+from fastapi import APIRouter, Depends, HTTPException
+from neon_data_models.enum import AccessRoles
 from neon_data_models.models.api.llm import LLMPersona
+from neon_data_models.models.user.database import PermissionsConfig
 from ovos_config.config import update_mycroft_config
 from ovos_utils.log import LOG
 
@@ -55,7 +57,10 @@ _hub_id = config.get("hub_id")
 if not _hub_id:
     _hub_id = generate_hub_id()
     LOG.info("Generated new hub_id: %s", _hub_id)
-    update_mycroft_config({"hana": {"hub_id": _hub_id}})
+    try:
+        update_mycroft_config({"hana": {"hub_id": _hub_id}})
+    except Exception as e:
+        LOG.error("Failed to persist hub_id to config: %s", e)
     config["hub_id"] = _hub_id
 
 
@@ -76,23 +81,37 @@ async def get_hub_identity() -> HubIdentityResponse:
     )
 
 
-@hub_route.post("/identity", dependencies=[Depends(jwt_bearer)])
+@hub_route.post("/identity")
 async def update_hub_identity(
     request: UpdateHubIdentityRequest,
+    token: str = Depends(jwt_bearer),
 ) -> HubIdentityResponse:
     """
     Update the display name of this Hub.
 
-    Requires authentication. The new display name is persisted to
-    the configuration file and survives container restarts.
+    Requires hub: ADMIN permission. The new display name is persisted
+    to the configuration file and survives container restarts.
     """
-    update_mycroft_config({"hana": {"hub_display_name": request.display_name}})
+    permissions = PermissionsConfig.from_roles(
+        jwt_bearer.client_manager.get_token_data(token).roles)
+    if permissions.hub < AccessRoles.ADMIN:
+        raise HTTPException(
+            status_code=403,
+            detail=f"Insufficient Hub management permission ({permissions.hub})",
+        )
+    warning = None
+    try:
+        update_mycroft_config({"hana": {"hub_display_name": request.display_name}})
+    except Exception as e:
+        LOG.error("Failed to persist hub_display_name to config: %s", e)
+        warning = "Display name updated in memory but could not be persisted to disk. It will revert on restart."
     config["hub_display_name"] = request.display_name
     LOG.info("Hub display_name updated to: %s", request.display_name)
     return HubIdentityResponse(
         hub_id=_hub_id,
         display_name=request.display_name,
         version=__version__,
+        warning=warning,
     )
 
 

--- a/neon_hana/schema/hub_requests.py
+++ b/neon_hana/schema/hub_requests.py
@@ -26,6 +26,7 @@
 
 from typing import Optional
 
+from neon_data_models.models.api.llm import LLMPersona
 from pydantic import BaseModel, Field, field_validator
 
 
@@ -41,12 +42,6 @@ class STTConfig(BaseModel):
     module: str = Field(description="Active STT plugin module name")
 
 
-class LLMConfig(BaseModel):
-    """LLM configuration."""
-
-    name: str = Field(description="Active LLM engine name")
-
-
 class HubConfigResponse(BaseModel):
     """Response model for GET /hub/config."""
 
@@ -58,7 +53,7 @@ class HubConfigResponse(BaseModel):
         default=None,
         description="Active STT configuration, or null if unavailable",
     )
-    llm: LLMConfig = Field(description="Active LLM configuration")
+    llm: LLMPersona = Field(description="Active LLM persona")
 
     model_config = {
         "json_schema_extra": {
@@ -66,7 +61,11 @@ class HubConfigResponse(BaseModel):
                 {
                     "tts": {"module": "neon-tts-plugin-coqui"},
                     "stt": {"module": "neon-stt-plugin-nemo"},
-                    "llm": {"name": "Neon Classic"},
+                    "llm": {
+                        "persona_name": "Neon Classic",
+                        "description": "OVOS parser-based assistant (no LLM)",
+                        "enabled": True,
+                    },
                 }
             ]
         }

--- a/neon_hana/schema/hub_requests.py
+++ b/neon_hana/schema/hub_requests.py
@@ -78,6 +78,7 @@ class HubIdentityResponse(BaseModel):
     hub_id: str = Field(description="Stable, human-readable hub identifier (e.g. 'bright-silver-falcon')")
     display_name: str = Field(description="User-configurable display name for this hub")
     version: str = Field(description="HANA software version")
+    warning: Optional[str] = Field(default=None, description="Non-fatal warning, e.g. config persistence failure")
 
     model_config = {
         "json_schema_extra": {
@@ -86,6 +87,7 @@ class HubIdentityResponse(BaseModel):
                     "hub_id": "bright-silver-falcon",
                     "display_name": "Neon Hub (Office)",
                     "version": "0.1.1a21",
+                    "warning": None,
                 }
             ]
         }

--- a/requirements/requirements.txt
+++ b/requirements/requirements.txt
@@ -5,6 +5,7 @@ pydantic~=2.5
 pyjwt~=2.8
 token-throttler~=1.4
 neon-mq-connector~=0.7,>=0.8.1a3
+neon-utils>=1.14.1a1
 ovos-config~=0.0,>=0.0.12
 ovos-utils~=0.0,>=0.0.38
 neon-data-models~=0.0,>=0.0.2a11

--- a/tests/test_app.py
+++ b/tests/test_app.py
@@ -683,7 +683,7 @@ class TestHanaApp(TestCase):
         data = response.json()
         self.assertEqual(data["tts"]["module"], "ovos-tts-plugin-mimic")
         self.assertEqual(data["stt"]["module"], "ovos-stt-plugin-vosk")
-        self.assertEqual(data["llm"]["name"], "Neon Classic")
+        self.assertEqual(data["llm"]["persona_name"], "Neon Classic")
 
     @patch("neon_hana.app.routers.hub._read_neon_yaml")
     def test_hub_config_defaults(self, mock_read):
@@ -704,7 +704,7 @@ class TestHanaApp(TestCase):
         data = response.json()
         self.assertIsNone(data["tts"])
         self.assertIsNone(data["stt"])
-        self.assertEqual(data["llm"]["name"], "Neon Classic")
+        self.assertEqual(data["llm"]["persona_name"], "Neon Classic")
 
     @patch("neon_hana.app.routers.hub._read_neon_yaml")
     def test_hub_config_key_without_module(self, mock_read):

--- a/tests/test_app.py
+++ b/tests/test_app.py
@@ -6,6 +6,7 @@ from unittest.mock import patch
 from fastapi.testclient import TestClient
 
 from neon_data_models.models.user import User
+from neon_data_models.models.user.database import PermissionsConfig
 
 _TEST_CONFIG = {
     "mq_default_timeout": 10,
@@ -50,6 +51,21 @@ class TestHanaApp(TestCase):
             self.tokens = response.json()
             self.assertIn("access_token", self.tokens, self.tokens)
         return self.tokens
+
+    @patch("neon_hana.mq_service_api.send_mq_request")
+    def _get_admin_tokens(self, send_request):
+        admin_user = User(
+            username="admin", password_hash="password",
+            permissions=PermissionsConfig(hub=30, core=20, diana=20,
+                                         node=20, llm=20))
+        send_request.return_value = {"user": admin_user.model_dump(),
+                                     "success": True}
+        response = self.test_app.post("/auth/login",
+                                      json={"username": "admin",
+                                            "password": "password"})
+        tokens = response.json()
+        self.assertIn("access_token", tokens, tokens)
+        return tokens
 
     def test_app_init(self):
         self.assertEqual(self.test_app.app.title, _TEST_CONFIG["fastapi_title"])
@@ -589,7 +605,7 @@ class TestHanaApp(TestCase):
         from neon_hana.app.dependencies import config
         self.addCleanup(config.pop, "hub_display_name", None)
 
-        token = self._get_tokens()["access_token"]
+        token = self._get_admin_tokens()["access_token"]
         original = self.test_app.get("/hub/identity").json()
 
         # Valid update
@@ -636,11 +652,21 @@ class TestHanaApp(TestCase):
             json={"display_name": "No Auth Hub"})
         self.assertIn(response.status_code, [401, 403], response.text)
 
+    @patch("neon_hana.mq_service_api.send_mq_request")
+    def test_hub_identity_update_insufficient_permissions(self, send_request):
+        # Guest user (hub: 0) should be rejected
+        token = self._get_tokens()["access_token"]
+        response = self.test_app.post(
+            "/hub/identity",
+            json={"display_name": "Unauthorized Hub"},
+            headers={"Authorization": f"Bearer {token}"})
+        self.assertEqual(response.status_code, 403, response.text)
+
     @patch("neon_hana.app.routers.hub.update_mycroft_config")
     @patch("neon_hana.mq_service_api.send_mq_request")
     def test_hub_identity_update_validation(self, send_request,
                                             mock_config_write):
-        token = self._get_tokens()["access_token"]
+        token = self._get_admin_tokens()["access_token"]
 
         # Empty display_name
         response = self.test_app.post(

